rust/stackablectl/src/cmds/debug.rs

```diff
@@ -353,6 +353,7 @@ impl AsyncRead for AsyncStdin {
     ) -> Poll<std::io::Result<()>> {
         loop {
             let mut ready = ready!(self.fd.poll_read_ready_mut(cx)?);
+            #[allow(clippy::blocks_in_conditions)]
```

Comment:
@Techassi I couldn't find a way to resolve this lint without it being arguably more complex with the addition of type definitions for the `r`.

Reply:
I will take a look.

Reply:
This seems to work:

```rust
loop {
    let mut ready = ready!(self.fd.poll_read_ready_mut(cx)?);

    let io = ready.try_io(|r| {
        let read = r.get_mut().read(buf.initialize_unfilled())?;
        buf.advance(read);
        Ok(())
    });

    break match io {
        Ok(res) => Poll::Ready(res),
        Err(_would_block) => {
            // Try to poll again, so that we re-register the waker
            continue;
        }
    };
}
```

Reply:
Thanks @Techassi, done in 85214f495b36cf42be67966518391b3e8ebcd820